### PR TITLE
Programmatic Management of Subscriptions

### PR DESCRIPTION
**Pull request by zugdud**
Based on:

https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#157-managing-subscriptions-programmatically

### CODE REVIEW

File: endpoints/makegoods.yaml [hunk 1]
```diff
@@ -247,212 +247,71 @@ paths:
...
+        summary: Enumerating subscriptions
+        description: Get a list of active subscriptions
+        operationId: Enumerating subscriptions
+        parameters:
+        - name: id
+          in: path
+          description: The subscription id
+          required: true
+        responses:
+          '202':
```
Comment (vadimi) [inline endpoints/makegoods.yaml]: if this is a synchronous request, we need to use `200`, not `202`

File: endpoints/orders.yaml [hunk 2]
```diff
@@ -290,229 +290,71 @@ paths:
...
+        summary: Enumerating subscriptions
+        description: Get a list of active subscriptions
+        operationId: Enumerating subscriptions
+        parameters:
+        - name: id
+          in: path
+          description: The subscription id
+          required: true
+        responses:
+          '202':
```
Comment (vadimi) [inline endpoints/orders.yaml]: if this is a synchronous request, we need to use `200`, not `202`

File: endpoints/schemas/commonSchemas.yaml [hunk 3]
```diff
@@ -56,6 +56,55 @@ components:
...
+        - id
+        - expirationDateTime
+      properties:
+        id:
+          description: Unique ID of the new subscription that can be used later to update/delete the subscription.
+          type: string
+        expirationDateTime:
+          description: The target web hook URL
+          type: string
+          format: date-time
```
Comment (vadimi) [inline endpoints/schemas/commonSchemas.yaml:88]: any chance we can specify the exact format of date-time? I'd recommend `ISO 8601`

File: endpoints/makegoods.yaml [hunk 4]
```diff
@@ -247,212 +247,71 @@ paths:
...
             required: true
             content:
               application/json:
                 schema:
-                  oneOf:
-                    - $ref: 'schemas/makegoodsSchemas.yaml#/components/schemas/makegoodsSubscribeSGS'
-                    - $ref: 'schemas/makegoodsSchemas.yaml#/components/schemas/makegoodsSubscribeSGSCounterParty'
+                  $ref: 'schemas/commonSchemas.yaml#/components/schemas/subscriptionRequest'
           responses:
             '202':
```
Comment (vadimi) [inline endpoints/makegoods.yaml]: if this is a synchronous request, we need to use `200`, not `202`

**Review by vadimi — CHANGES_REQUESTED** [review]

File: endpoints/orders.yaml [hunk 5]
```diff
@@ -290,229 +290,71 @@ paths:
...
             required: true
             content:
               application/json:
                 schema:
-                  oneOf:
-                    - $ref: 'schemas/ordersSchemas.yaml#/components/schemas/ordersSubscribeSGS'
-                    - $ref: 'schemas/ordersSchemas.yaml#/components/schemas/ordersSubscribeSGSCounterParty'
+                  $ref: 'schemas/commonSchemas.yaml#/components/schemas/subscriptionRequest'
           responses:
             '202':
```
Comment (vadimi) [inline endpoints/orders.yaml]: same here I guess, `200`, not `202`

Comment (Spintronic) [reply endpoints/orders.yaml]: should this be 201 as noted in common schema description?

File: endpoints/orders.yaml [hunk 6]
```diff
@@ -290,229 +290,71 @@ paths:
...
+        summary: Remove a subscription
+        description: Existing subscriptions can be deleted by making a DELETE request against the subscription resource
+        operationId: Remove a subscription
+        parameters:
+        - name: id
+          in: path
+          description: The subscription id
+          required: true
+        responses:
+          '204':
```
Comment (Spintronic) [inline endpoints/orders.yaml]: synchronous delete? 200 here?

File: endpoints/orders.yaml [hunk 7]
```diff
@@ -290,229 +290,71 @@ paths:
...
-                    $ref: 'schemas/commonSchemas.yaml#/components/schemas/Error'
\ No newline at end of file
+                  $ref: 'schemas/commonSchemas.yaml#/components/schemas/Error'
+      
+      get:
+        summary: Enumerating subscriptions
+        description: Get a list of active subscriptions
+        operationId: Enumerating subscriptions
+        parameters:
+        - name: id
```
Comment (Spintronic) [inline endpoints/orders.yaml]: not sure supplying subscription id makes sense. thought the immediate use case of enumerate subscriptions was to get the ids to facilitate a delete.

File: endpoints/makegoods.yaml (hunk 4 above)
Comment (Spintronic) [reply endpoints/makegoods.yaml]: should it be 201 as per description?

File: endpoints/makegoods.yaml [hunk 8]
```diff
@@ -247,212 +247,71 @@ paths:
...
+        summary: Remove a subscription
+        description: Existing subscriptions can be deleted by making a DELETE request against the subscription resource
+        operationId: Remove a subscription
+        parameters:
+        - name: id
+          in: path
+          description: The subscription id
+          required: true
+        responses:
+          '204':
```
Comment (Spintronic) [inline endpoints/makegoods.yaml]: 200 for synchronous delete?

File: endpoints/makegoods.yaml [hunk 9]
```diff
@@ -247,212 +247,71 @@ paths:
...
-                    $ref: 'schemas/commonSchemas.yaml#/components/schemas/Error'
\ No newline at end of file
+                  $ref: 'schemas/commonSchemas.yaml#/components/schemas/Error'
+      
+      get:
+        summary: Enumerating subscriptions
+        description: Get a list of active subscriptions
+        operationId: Enumerating subscriptions
+        parameters:
+        - name: id
```
Comment (Spintronic) [inline endpoints/makegoods.yaml]: suggest removing id parameter. thought enumerate would return list of ids to facilitate deletes.

File: endpoints/schemas/commonSchemas.yaml (hunk 3 above)
Comment (Spintronic) [reply endpoints/schemas/commonSchemas.yaml:88]: @vadimi It appears that the YAML standard for dates uses the ISO-8601 standard. https://symfony.com/doc/current/components/yaml/yaml_format.html

**Review by vadimi — APPROVED** [review]

**Pull request merged**